### PR TITLE
Don't pack symlinks for wheel

### PR DESCRIPTION
**Pull request by Wovchena**

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: src/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -51,16 +51,23 @@ target_link_libraries(${TARGET_NAME} PUBLIC openvino::runtime PRIVATE nlohmann_j
...
     ARCHIVE_OUTPUT_DIRECTORY "$<1:${CMAKE_BINARY_DIR}/openvino_genai/>"
     LIBRARY_OUTPUT_DIRECTORY "$<1:${CMAKE_BINARY_DIR}/openvino_genai/>"
     RUNTIME_OUTPUT_DIRECTORY "$<1:${CMAKE_BINARY_DIR}/openvino_genai/>"
 )
+# Extract two last digits from CMAKE_PROJECT_VERSION_MAJOR because SOVERSION can only contain up to 4 symbols.
+string(REGEX MATCH [=[[0-9][0-9]$]=] MAJOR_SUFFIX ${CMAKE_PROJECT_VERSION_MAJOR})
+if(SKBUILD AND NOT WIN32)
+    # Don't pack symlinks but append version hash to the name for wheel
+    set_target_properties(${TARGET_NAME} PROPERTIES
+        SUFFIX .so.${MAJOR_SUFFIX}${CMAKE_PROJECT_VERSION_MINOR}${CMAKE_PROJECT_VERSION_PATCH})
```
Comment (ilya-lavrenov) [inline src/cpp/CMakeLists.txt:65]: `.so` is not valid for macOS

Comment (Wovchena) [reply src/cpp/CMakeLists.txt:65]: Updated

File: src/cpp/CMakeLists.txt [hunk 2]
```diff
@@ -59,10 +59,13 @@ set_target_properties(${TARGET_NAME} PROPERTIES
...
 # Extract two last digits from CMAKE_PROJECT_VERSION_MAJOR because SOVERSION can only contain up to 4 symbols.
 string(REGEX MATCH [=[[0-9][0-9]$]=] MAJOR_SUFFIX ${CMAKE_PROJECT_VERSION_MAJOR})
-if(SKBUILD AND NOT WIN32)
+if(SKBUILD AND LINUX)
     # Don't pack symlinks but append version hash to the name for wheel
     set_target_properties(${TARGET_NAME} PROPERTIES
         SUFFIX .so.${MAJOR_SUFFIX}${CMAKE_PROJECT_VERSION_MINOR}${CMAKE_PROJECT_VERSION_PATCH})
+elseif(SKBUILD AND APPELE)
+    set_target_properties(${TARGET_NAME} PROPERTIES
+        SUFFIX .${MAJOR_SUFFIX}${CMAKE_PROJECT_VERSION_MINOR}${CMAKE_PROJECT_VERSION_PATCH}.dylib)
```
Comment (ilya-lavrenov) [inline src/cpp/CMakeLists.txt:68]:
```suggestion
        SUFFIX .${MAJOR_SUFFIX}${CMAKE_PROJECT_VERSION_MINOR}${CMAKE_PROJECT_VERSION_PATCH}${CMAKE_SHARED_LIBRARY_SUFFIX})
```

File: src/cpp/CMakeLists.txt [hunk 3]
```diff
@@ -59,10 +59,13 @@ set_target_properties(${TARGET_NAME} PROPERTIES
 )
 # Extract two last digits from CMAKE_PROJECT_VERSION_MAJOR because SOVERSION can only contain up to 4 symbols.
 string(REGEX MATCH [=[[0-9][0-9]$]=] MAJOR_SUFFIX ${CMAKE_PROJECT_VERSION_MAJOR})
-if(SKBUILD AND NOT WIN32)
+if(SKBUILD AND LINUX)
     # Don't pack symlinks but append version hash to the name for wheel
     set_target_properties(${TARGET_NAME} PROPERTIES
         SUFFIX .so.${MAJOR_SUFFIX}${CMAKE_PROJECT_VERSION_MINOR}${CMAKE_PROJECT_VERSION_PATCH})
```
Comment (ilya-lavrenov) [inline src/cpp/CMakeLists.txt:65]:
```suggestion
        SUFFIX ${CMAKE_SHARED_LIBRARY_SUFFIX}.${MAJOR_SUFFIX}${CMAKE_PROJECT_VERSION_MINOR}${CMAKE_PROJECT_VERSION_PATCH})
```

File: src/cpp/CMakeLists.txt (hunk 2 above)
Comment (Wovchena) [reply src/cpp/CMakeLists.txt:68]: If it's not required to fit into 4 symbols, should it be just the project version?

Comment (Wovchena) [reply src/cpp/CMakeLists.txt:68]: Oh, never mind

**Pull request merged**